code/headers/servo_c.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <pwm_s.hpp>
+
+namespace r2d2::servo_interface {
+    class servo_c {
+    protected:
+        /**
+         * the current angle the servo is at
+         * this value can be anywhere between -360 to +360
+         */
+        int16_t angle;
+
+        /**
+         * the pwm needed to put the servo at the desired angle
+         */
+        pwm_s pwm;
+
+    public:
+        /**
+         * sets the angle and the pwm accordingly to the servo
+         *
+         * @param new_angle - the new angle of the servo
```

Comment:
maybe nice to include in the comment the range that the new angle can have. I persanily also prefer to tell the units used. I persume the angle has to be given in degrees, but maybe you chose radians how chould I know? If you agree, take a look of the rest of you comments as wel.